posthog/src/main/java/com/posthog/java/PostHog.java

```diff
@@ -43,30 +39,32 @@ public PostHog build() {
     private PostHog(Builder builder) {
         apiKey = builder.apiKey;
         host = builder.host;
+        sender = new Sender(apiKey, host);
+        startQueueManager();
     }
 
-    private void send(String distinctId, String event, HashMap<String, Object> properties) {
-        JSONObject eventJson = getEventJson(event, distinctId, properties);
-        String json = getRequestBody(Collections.singletonList(eventJson));
-
-        System.out.println(json);
-
+    public void shutdown() {
+        queueManager.stop();
         try {
-            OkHttpClient client = new OkHttpClient();
-
-            MediaType JSON = MediaType.parse("application/json; charset=utf-8");
-            RequestBody body = RequestBody.create(json, JSON);
-
-            Request request = new Request.Builder().url(host + "/batch").post(body).build();
-            System.out.println(request);
-            Call call = client.newCall(request);
-            Response response = call.execute();
-            System.out.println(response.body().string());
-        } catch (Exception e) {
-            System.out.println(e);
+            queueManagerThread.join(); // wait for the current items in queue to be sent
+        } catch (InterruptedException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
         }
     }
 
+    private void startQueueManager() {
+        queueManager = new QueueManager(sender);
+        queueManagerThread = new Thread(queueManager, "PostHog QueueManager thread");
+        queueManagerThread.start();
+        // TODO handle interrupts? (via addShutdownHook)
```

Comment:
yeah let's try to send on shutdown

Reply:
I wasn't sure yet of all the implications of adding `addShutdownHook` to call `shutdown` & whether we want that, was going to look into it.

Reply:
what I'd like to see is just us trying to send requests if we come crashing down